multichain-api-gateway/src/lib.rs

```diff
@@ -45,13 +46,31 @@ impl From<BlockscoutSettings> for ApiEndpoints {
 }
 
 impl ApiEndpoints {
-    async fn make_requests(self, query: &str) -> Vec<(Instance, String)> {
+    async fn make_requests(
+        self,
+        query: &str,
+        method: &Method,
+        body: Bytes,
+        headers: &HeaderMap,
+    ) -> Vec<(Instance, String)> {
         let client = Client::new();
 
         stream::iter(self.apis)
             .map(|(instance, mut url)| async {
                 url.set_query(Some(query));
-                let resp = client.get(url).send().await.unwrap();
+
+                let resp = client
+                    .request(method.clone(), url)
+                    .header(
+                        "Content-Type",
+                        headers
+                            .get("Content-Type")
```

Comment:
hm, but what if I would like to pass other headers? Like `Authorization: <auth_token>`

I see that `headers()` accepts `http::HeaderMap`, however we have `actix_web::http::header::HeaderMap`, it's uncomfortable(((
but maybe we have convert one type to another? what do you think? 👀